src/proto/single/cmd.rs

```diff
@@ -311,3 +313,39 @@ impl<'a, S: AsRef<str>> QueueControl<'a, S> {
         Self { action, queues }
     }
 }
+
+// ---------------------- MUTATE -------------------
+
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize)]
+#[serde(rename_all = "lowercase")]
+#[non_exhaustive]
+pub(crate) enum MutationType {
+    #[default]
+    Kill,
+    Requeue,
+    Discard,
+    Clear,
+}
+
+fn filter_is_empty(f: &Option<&MutationFilter<'_>>) -> bool {
+    // Rust 1.82 has got the genious `Option::is_none_or`,
+    // and `Option::is_some_and` which will allow us to:
+    // ```rust
+    // f.is_none_or(|f| f.is_empty())
+    // ```
+    // As of crate version `0.13.1-rc0`, we've got `1.70` as MSRV.
+    match f {
+        None => true,
+        Some(f) => f.is_empty(),
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
+pub(crate) struct MutationAction<'a> {
+    pub(crate) cmd: MutationType,
+    pub(crate) target: MutationTarget,
+    #[serde(skip_serializing_if = "filter_is_empty")]
```

Comment:
Is an empty-but-set mutation filter actually == to an unset filter? In some systems, the two have different semantics (usually, unset means "use the default").